receiver/prometheusremotewritereceiver/receiver_test.go

```diff
@@ -131,6 +132,85 @@ func TestHandlePRWContentTypeNegotiation(t *testing.T) {
 	}
 }
 
+func TestHandlerPRWContentTypeNegotiation_httptest(t *testing.T) {
+	body := writev2.Request{}
+	pBuf := proto.NewBuffer(nil)
+	err := pBuf.Marshal(&body)
+	assert.NoError(t, err)
+
+	var compressedBody []byte
+	snappy.Encode(compressedBody, pBuf.Bytes())
+
+	// no content type
+	expectedCode := http.StatusUnsupportedMediaType
+
+	req := httptest.NewRequest(http.MethodPost, "/api/v1/write", bytes.NewBuffer(compressedBody))
+	req.Header.Set("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "")
+	w := httptest.NewRecorder()
```

Comment:
@ArthurSens It's not done yet. It's just an idea. Did you see any problem to use `httptest` instead of call `setupServer`? Like you did here ⬇️

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d74ba549cdedb249a58b26ea39fa1644e3786d1e/receiver/prometheusremotewritereceiver/receiver_test.go#L74-L75

httptest take care of a bunch of things. Including the creation of a valid base url and ports. For that specific case where we are just validating the expected `statusCode` given a header, I think that would be a good approach to avoid the `flaky test thing`. WDYT?

Reply:
@pjanotti some constraint about this approach?

Reply:
Yes, I like your approach! Let's do it!